File: yologram-api-v2/app/infra/client/pms/pms_api_client.py
# 타 도메인 경계 클라이언트 — 타 도메인 리포지토리 import는 app/infra/client에서만 허용 (domain 간 직접 참조 금지).
# MSA 분리 시 Local 구현 대신 Rest(HTTP) 구현을 추가해 교체한다.
from typing import Protocol

from sqlalchemy.orm import Session

from app.domain.pms.tech.repository import TechPostRepository


class PmsApiClient(Protocol):
    """
    tech comment → tech post 도메인 경계 호출 추상화 (대상 글 존재 검증).
    모놀리식에서는 tech post 리포지토리를 직접 호출(LocalPmsApiClient),
    MSA 분리 시 post-api HTTP 호출 구현으로 교체한다.
    """

    def exists(self, post_id: int) -> bool:
        """post_id의 게시글이 존재하면 True."""
        ...


class LocalPmsApiClient:

    def __init__(self, db: Session):
        self.repository = TechPostRepository(db)

    def exists(self, post_id: int) -> bool:
        return self.repository.find_by_id(post_id) is not None

File: yologram-worker/src/main/kotlin/link/yologram/worker/infra/client/pms/PmsApiClient.kt
package link.yologram.worker.infra.client.pms

/**
 * pms 도메인 경계 클라이언트 — 현재는 같은 DB를 읽는 Local 구현이지만,
 * MSA로 분리하면 이 인터페이스만 Rest 구현으로 갈아끼운다 (api-v1 infra/client 규칙 미러).
 */
interface PmsApiClient {

    /** id 범위의 게시글을 카운트·카테고리와 함께 조회 (인덱싱용, 없는 id는 결과에서 빠진다) */
    fun findPostsForIndex(from: Long, to: Long): List<TechPostForIndex>
}
